pkg/labels/array.go

```diff
@@ -96,13 +96,29 @@ nextLabel:
 	return true
 }
 
+// Intersects returns true if ls contains at least one label in needed.
+//
+// This has the same matching semantics as Has, namely,
+// ["k8s:foo=bar"].Intersects(["any:foo=bar"]) == true
+// ["any:foo=bar"].Intersects(["k8s:foo=bar"]) == false
+func (ls LabelArray) Intersects(needed LabelArray) bool {
```

Comment:
Should there be a mention here that it's desirable for the label arrays to be sorted to take advantage of efficiency and warn the caller that if not sorted, then it can be expensive?

Reply:
(LabelArrays are required to be sorted, so that's not an issue).

The sorting requirement is moot anyways, since it doesn't work for CIDR expansion :-/. Though that does make me want to be clever and use binary search in `.Has()` now... Task for another PR.